modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -690,13 +691,7 @@ private void updateAssignmentInternal(ConfigurationNotificationEvent<byte[]> ass
                                                 newPartAssignment,
                                                 new PartitionListener(
                                                     partitionStorage,
-                                                    // TODO: https://issues.apache.org/jira/browse/IGNITE-17579 TxStateStorage management.
-                                                    new TxStateRocksDbStorage(
-                                                        Paths.get("tx_state_storage" + tblId + partId),
-                                                        Executors.newSingleThreadScheduledExecutor(),
-                                                        Executors.newFixedThreadPool(1),
-                                                        () -> 1000
-                                                    ),
+                                                    internalTbl.txnStateStorage().getOrCreateTxnStateStorage(partId),
```

Comment:
I see that groupOptionsForPartition is not fixed, why have you decided to do that separately?